python/perspective/perspective/src/accessor.cpp

```diff
@@ -74,14 +74,10 @@ infer_type(t_val x, t_val date_validator) {
     } else if (py::isinstance<py::bool_>(x) || type_string == "bool") {
         // booleans are both instances of bool_ and int_ -  check for bool first
         t = t_dtype::DTYPE_BOOL;
+    } else if (py::isinstance<py::float_>(x)) {
+        t = t_dtype::DTYPE_FLOAT64;
     } else if (py::isinstance<py::int_>(x)) {
-        double x_float64 = x.cast<double>();
-        if ((std::fmod(x_float64, 1.0) == 0.0) && (x_float64 < 10000.0)
-            && (x_float64 != 0.0)) {
-            t = t_dtype::DTYPE_INT32;
-        } else {
-            t = t_dtype::DTYPE_FLOAT64;
-        }
+        t = t_dtype::DTYPE_INT32;
```

Comment:
Do we have an overflow test? In python 3 ints are int64s

Reply:
added overflow test - promotion takes care of overflow